test/e2e/streaming_test.go

```diff
@@ -78,6 +77,45 @@ func (suite *StreamingTestSuite) TestStreaming() {
 	ProductionSmokeTest("simple-streaming")
 }
 
+func (suite *StreamingTestSuite) TestStreamingWithTLS() {
+	if !usingOLM {
+		t.Skip("This test should only run when using OLM")
+	}
+	// Make sure ES and the kafka instance are available
+	waitForElasticSearch()
+	waitForKafkaInstance()
+
+	kafkaUserName := "my-user"
+	kafkaUser := getKafkaUser(kafkaUserName, kafkaNamespace)
```

Comment:
Does it still work if you move this to before the call to `waitForKafkaInstance()` ? Reason being that there's a (small) chance that the `KafkaUser` resource might not have been created (or reconciled) yet by the time it's secret is used by the Jaeger instance. If you can move it to before the Kafka Instance, then the chances are really, really small, as it has all the time it takes for a Kafka instance to boot to create the resource and secrets.

Reply:
It would most likely work, since this is the second test, and the kafka instance would have to be ready for the first test to have even run.  :-)  So I could move it, but I doubt that would make much difference.

Is there anyway to confirm programmatically that all of the user resources are ready?  If not I'd rather just add a short sleep after creating the user.

Reply:
When you create a KafkaUser, Strimzi will create the secret that we require on the next reconciliation. So, you can just check whether the secret has been created already.

Or, you can just do "nothing" and wait for the deployment to eventually succeed, as it will not be created until all the requirements (like secrets) are available. The disadvantage of that is, if the test fails, you'll get a better error message if you are checking for the secret's existence.